### PR TITLE
Delivery Packages (part 1)

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
#### Summary

```SUMMARY: Content "Adds delivery packages."```

#### Purpose of change

This adds delivery packages of which can be unpacked to reveal random goodies. It's like Christmas any day of the year!

This is a revival of [what I started roughly 1.5 years ago](https://github.com/CleverRaven/Cataclysm-DDA/issues/25194) and will be split into several parts.

Packages come in 3 sizes like the cardboard boxes. Their contents are based on their size, and you can find the list of items I made before in the link above. But due to the fact I made the package content items groups 1.5 years ago, I may be missing some obvious candidates for package contents that have been added to the game since then. So, if you have any suggestions of additional package contents, let me know and I can add them. **300+ DIFFERENT PACKAGE ITEM GROUPS!!!**

Changes in this pr:
- [x] adjusts cardboard box items to work for packages
- [x] adds package items

#### Additional Context

Changes in following prs:
- add package item groups
- add unpack/disassemble action to packages
- add packages to locations
- add a mail truck
- add 1-2 delivery trucks
- add package distribution center (2x3) and package distribution hub (1x1) locations

### CODE REVIEW

**Comment by Fris0uman** [comment]
You need to lint your changes, you can use this http://dev.narc.ro/cataclysm/format.html

**Comment by cosmiccoincidence** [comment]
> You need to lint your changes, you can use this http://dev.narc.ro/cataclysm/format.html

Yeah, that is what I used. But I did reconfigure some of the short lists at the top of /itemgroups/packages.json as they are easier to read vertically than horizontally.

**Comment by AMurkin** [comment]
[Automated checks disagree](https://github.com/CleverRaven/Cataclysm-DDA/pull/38155/checks?check_run_id=455306518#step:4:17) with this logic.
So yeah, do what the linter says.

**Comment by curstwist** [comment]
I've got some boxes I've used in houses and basements, they have general house spawns in them mostly.  Be cool to move them into this packaging system eventually.  So, I'd suggest household junk boxes, they type of stuff you store and forget about for 20 years, maybe even holiday decorations and such if you want to get fancy.

File: data/json/items/fluff.json [hunk 1]
```diff
@@ -55,5 +55,44 @@
     "material": [ "paper" ],
     "symbol": "*",
     "color": "light_gray"
+  },
+  {
+    "id": "package_small",
+    "type": "GENERIC",
+    "category": "other",
+    "name": { "str": "small package", "str_pl": "small packages" },
+    "description": "A sealed, small-size cardboard box, it may have something inside.",
```
Comment (matskuman5) [inline data/json/items/fluff.json]:
```suggestion
    "description": "A sealed, small-size cardboard box.  It may have something inside.",
```

Comment (matskuman5) [reply data/json/items/fluff.json]: same for the other descriptions

**Comment by cosmiccoincidence** [comment]
> I've got some boxes I've used in houses and basements...

@curstwist I believe I made some item groups of the same fashion somewhere... We didn't have cardboard boxes at the times so I used crates, and the crates would spawn living room/kitchen/garage/etc stuff depending on what room. They were mostly only used in the 'vacant homes' pack I did which included homes that had garage sales, were about to move out, just moved in, etc.

I agree it would maybe be nice to consolidate all this into the new system at some point in the future.

**Comment by curstwist** [comment]
>  room. They were mostly only used in

ah right, I have seen those.  I'll take a look at their groups.

File: data/json/itemgroups/packages.json [hunk 2]
```diff
@@ -0,0 +1,2567 @@
...
+    "type": "item_group",
+    "id": "package_small_clothing_underwear9",
+    "subtype": "collection",
+    "items": [ [ "under_armor" ], [ "under_armor_shorts" ] ]
+  },
+  {
+    "type": "item_group",
+    "id": "package_small_clothing_underwear10",
+    "subtype": "collection",
+    "items": [ [ "long_undertop" ], [ "long_underpants" ] ]
```
Comment (BevapDin) [inline data/json/itemgroups/packages.json]: There are numbers missing. Each entry of the "items" array should be an array with **two** elements: the item id and the probability.
```suggestion
    "items": [ [ "long_undertop", 100 ], [ "long_underpants", 100 ] ]
```
This applies to all the other "item_group" object here as well.

**Comment by Little-119** [comment]
This is a prime opportunity for a movie reference. Add a ~~Wilson~~ William brand volleyball to the list of possible items

**Comment by cosmiccoincidence** [comment]
> volleyball

There doesn't seem to be one in-game. Though in the future it would be easy to add as well as a park location with a sandy volleyball court.

**Comment by I-am-Erk** [comment]
This is too much for a single PR. Can you keep the list of packages down a bit, and add more packages in another PR?

It seems to me packages are something that could use the disassembly script like MREs, rather than just appearing in itemgroups

**Comment by cosmiccoincidence** [comment]
> This is too much for a single PR. Can you keep the list of packages down a bit, and add more packages in another PR?

Uhm, I guess. The only thing removing some item groups from this PR would do is lower line count. I admit I had some more item groups ready to push and add to this PR but Ill wait and spread things out more if that's needed.

> It seems to me packages are something that could use the disassembly script like MREs, rather than just appearing in itemgroups

I plan to use the 'unpack' being moved to json in #38111 (which the gobag uses)

**Comment by cosmiccoincidence** [comment]
Breaking this PR into more pieces as requested by Erk.

**Comment by cosmiccoincidence** [comment]
Closing this until I have time to come back to it.

**Pull request closed without merging**